### ISSUE TITLE
RUF022: allow changing `__all__` sorting to follow definition order

### ISSUE THREAD

**Issue by Jackenmen** [astral-sh/ruff#20952]
### Summary

I have a suggestion to extend the `__all__` sorting a bit and allow sorting by the *definition* order of names (functions/classes/variables) that are put in `__all__`.

So, for example:
```py
from .submodule import reexported_name
from .submodule_b import different_reexported_name

__all__ = (
    "reexported_name",
    "different_reexported_name",
    "function_b",
    "X",
    "function_a",
    "function_c",
    "Y",
)


def function_b():
    ...


class X:
   ...


def function_a():
    ...


def function_c():
    ...


class Y:
   ...
```

The reason for such a suggestion is that I think it makes it easier to find out what names of public APIs are missing in `__all__` and what names shouldn't be in it (especially the latter one) since it allows one to just go through the file and look through the names in `__all__` in the same order.

---

I've based this enhancement request on my original request for this rule: https://github.com/astral-sh/ruff/issues/1198#issuecomment-1509734136. I was configuring Ruff for a new project today and realised that this part never became a thing.
